"pathspace" options, vtysh-suid-cleanups, "vty_frame()"

eqvinox opened the pull request:
2010 says hello and sends patches...

donaldsharp commented:
I think that the SA has a point about ignoring the return values for seteuid and setegid.

eqvinox commented:
> I think that the SA has a point about ignoring the return values for seteuid and setegid.

It does. fixed.

> Don't we need to insert a prefix into the non-integrated config file name as well?

That happens in `frr_init`:
```
	snprintf(config_default, sizeof(config_default), "%s%s/%s%s.conf",
		 frr_sysconfdir, p_pathspace, di->name, p_instance);
```

> We need to document the -N option somewhere.

=> frr-args(8)

> Interesting. We added code to vtysh to drop empty interface blocks. This does it at the daemon level

It can do more than that actually - it can be used to print default values "but only if the block is present anyway." It used to do exactly that for `ip nd suppress-ra`, but we're not printing that anymore in any case.
